### PR TITLE
WR-30 CI Test and SwiftLint setup

### PR DESCRIPTION
**Pull request by kaulex99**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      - '.swiftlint.yml'
+      - '**/*.swift'
+
+jobs:
+  SwiftLint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: GitHub Action for SwiftLint
+        uses: norio-nomura/action-swiftlint@3.2.1
```
Comment (dev-Fabi) [inline .github/workflows/lint.yml:16]: Does this also check formatting (swiftformat)?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup Ruby 3.2.1
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2.1'
+          bundler-cache: true
+      
+      - name: Netrc api.github.com
```
Comment (dev-Fabi) [inline .github/workflows/test.yml:19]:
```suggestion
      - name: Netrc maven.pkg.github.com
```
Seems as I forgot to update this in the publish workflow

File: project.yml [hunk 3]
```diff
@@ -130,4 +138,29 @@ targets:
...
+    type: bundle.ui-testing
+    platform: iOS
+    sources:
+      - WaiterRobotUITests
+    dependencies:
+      - target: WaiterRobotLava
+    settings:
+      base:
+        GENERATE_INFOPLIST_FILE: true
+        IPHONEOS_DEPLOYMENT_TARGET: '15.0'
```
Comment (dev-Fabi) [inline project.yml:165]: This is already defined as base (line 26), so it is not needed here again, right?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+name: Run unit and ui tests
+
+on:
+  push:
```
Comment (dev-Fabi) [inline .github/workflows/test.yml:4]:
```suggestion
  push:
    branches-ignore:
      - 'develop'
      - 'main'
```

For this branches a push triggers a release. The execution of the test should be added to the publish workflow.
(also see the android test and publish workflow)

File: .github/workflows/test.yml (hunk 2 above)
Comment (dev-Fabi) [reply .github/workflows/test.yml:22]: What about that? 😅

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+          password: ${{ secrets.NETRC_TOKEN }}
+    
+      - name: Setup xcodegen and init xcode project
+        uses: xavierLowmiller/xcodegen-action@1.1.2
+        with:
+          spec: project.yml
+          use-cache: true
+
+      - name: Run tests with fastlane
+        run: bundle exec fastlane test
```
Comment (dev-Fabi) [inline .github/workflows/test.yml:36]: This should now also be added to the publish workflow (right before starting the release)

**Review by dev-Fabi — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (kaulex99) [reply .github/workflows/test.yml:22]: this is already fixed 🤔

File: .github/workflows/test.yml (hunk 5 above)
Comment (kaulex99) [reply .github/workflows/test.yml:36]: will do in separate PR

File: .github/workflows/test.yml (hunk 2 above)
Comment (dev-Fabi) [reply .github/workflows/test.yml:22]: No, the command is correct, but the name of the lane is outdated 😅

**Pull request merged**